vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -745,8 +754,12 @@ class OnboardingViewModel @AssistedInject constructor(
         return loginConfig?.homeServerUrl
     }
 
-    fun getSsoUrl(redirectUrl: String, deviceId: String?, providerId: String?): String? {
-        return authenticationService.getSsoUrl(redirectUrl, deviceId, providerId)
+    fun getSsoUrl(redirectUrl: String, deviceId: String?, provider: SsoIdentityProvider?): String? {
```

Comment:
Interestingly I think changing the name `fetchSsoUrl` would justify the side effect, since fetch communicates that it's more than just a simple getter

Reply:
agreed! https://github.com/vector-im/element-android/pull/6042/commits/db3cb42cc341a55bb3c015f2164402bcd2d5f566